docs: add section to adjust cmp label and menu length (#165)
AstroNvim/docs: src/content/docs/recipes/cmp.mdx

  end,
}
```

### Limit Label and Menu Item Length

To limit the label and menu item length and prevent the documentation window from getting too small, you can use the following configuration:

```lua title="lua/plugins/cmp.lua"
return {
  "hrsh7th/nvim-cmp",
  opts = function(_, opts)
    local astrocore, astroui = require "astrocore", require "astroui"
    if not opts.formatting then opts.formatting = {} end
    opts.formatting.format = astrocore.patch_func(opts.formatting.format, function(format, ...)
      -- get item from original formatting function
      local vim_item = format(...)

      -- truncate label to maximum of 25% of the window
      local label = vim_item.abbr
      local truncated_label = vim.fn.strcharpart(label, 0, math.floor(0.25 * vim.o.columns))
      if truncated_label ~= label then vim_item.abbr = truncated_label .. astroui.get_icon "Ellipsis" end

      -- truncate menu to maximum of 25% of the window

>       local menu = vim_item.menu or ""
>       local truncated_menu = vim.fn.strcharpart(menu, 0, math.floor(0.25 * vim.o.columns))
>       if truncated_menu ~= menu then vim_item.menu = truncated_menu .. astroui.get_icon "Ellipsis" end

Reviewer: ```suggestion
      if vim_item.menu then
        local menu = vim_item.menu
        local truncated_menu = vim.fn.strcharpart(menu, 0, math.floor(0.25 * vim.o.columns))
        if truncated_menu ~= menu then vim_item.menu = truncated_menu .. astroui.get_icon "Ellipsis" end
      end
```

We should only do the truncation stuff if the menu field exists rather than treating it as an empty string

  end,
}
```

### Limit Label and Menu Item Length

To limit the label and menu item length and prevent the documentation window from getting too small, you can use the following configuration:

```lua title="lua/plugins/cmp.lua"
return {
  "hrsh7th/nvim-cmp",
  opts = function(_, opts)
    local astrocore, astroui = require "astrocore", require "astroui"

>     if not opts.formatting then opts.formatting = {} end
>     opts.formatting.format = astrocore.patch_func(opts.formatting.format, function(format, ...)
>       -- get item from original formatting function
>       local vim_item = format(...)
> 
>       -- truncate label to maximum of 25% of the window
>       local label = vim_item.abbr
>       local truncated_label = vim.fn.strcharpart(label, 0, math.floor(0.25 * vim.o.columns))
>       if truncated_label ~= label then vim_item.abbr = truncated_label .. astroui.get_icon "Ellipsis" end
> 
>       -- truncate menu to maximum of 25% of the window
>       if vim_item.menu then
>         local menu = vim_item.menu
>         local truncated_menu = vim.fn.strcharpart(menu, 0, math.floor(0.25 * vim.o.columns))
>         if truncated_menu ~= menu then vim_item.menu = truncated_menu .. astroui.get_icon "Ellipsis" end
>       end
> 
>       return vim_item
>     end)

Reviewer: ```suggestion
    local function truncate(str, len)
      if not str then return end
      local truncated = vim.fn.strcharpart(str, 0, len)
      return truncated == str and str or truncated .. astroui.get_icon "Ellipsis"
    end

    if not opts.formatting then opts.formatting = {} end
    opts.formatting.format = astrocore.patch_func(opts.formatting.format, function(format, ...)
      -- get item from original formatting function
      local vim_item = format(...)

      -- truncate text fields to maximum of 25% of the window
      vim_item.abbr = truncate(vim_item.abbr, math.floor(0.25 * vim.o.columns))
      vim_item.menu = truncate(vim_item.menu, math.floor(0.25 * vim.o.columns))

      return vim_item
    end)
```

We might be able to decrease some duplicate code and also make it a bit easier to read. This is definitely super nitpicky, so feel free to ignore this. Also technically could be mildly optimized by only calculating the max width once but I figured the user might want to easily be able to tweak the 2 truncation lengths

  end,

> }

Reviewer: ````suggestion
}
```
````

there is just the closing backticks missing here
